File: csharp/MkCert/Encrypt.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Org.BouncyCastle.Crypto.Parameters;
using Org.BouncyCastle.Crypto.Digests;
using Org.BouncyCastle.Crypto.Engines;
using Org.BouncyCastle.Crypto.Encodings;

namespace MkCert
{
    public class Encrypt
    {
        public static byte[] EncryptBytes(RsaKeyParameters publicKey, byte[] content)
        {
            var rsaEngine = new RsaEngine();
            
            var oaepEncoding = new OaepEncoding(rsaEngine,new Sha256Digest(), new Sha256Digest(),null); // MGF1-SHA256
            oaepEncoding.Init(true, publicKey);
            var inputBlockSize = oaepEncoding.GetInputBlockSize();
            using (System.IO.MemoryStream ms = new System.IO.MemoryStream())
            {
                
                System.IO.BinaryWriter bw = new System.IO.BinaryWriter(ms);
                
                int i = 0;
                for (i = 0; i < content.Length; i += inputBlockSize)
                {
                    var ciphertext = oaepEncoding.ProcessBlock(content, i, i + inputBlockSize < content.Length ? inputBlockSize : content.Length-i );
                    bw.Write(ciphertext);
                }
                bw.Flush();
                return ms.ToArray();
            }
        }

        public static string EncryptBytesAndEncodeWithBase64(RsaKeyParameters publicKey, byte[] content)
        {
            var cipherBytes = EncryptBytes(publicKey, content);
            return Org.BouncyCastle.Utilities.Encoders.Base64.ToBase64String(cipherBytes);
        }
        
    }
}

File: java/src/main/java/Encrypt.java
/**
 * Created by beeven on 12/20/2016.
 */

import org.bouncycastle.crypto.InvalidCipherTextException;
import org.bouncycastle.crypto.digests.SHA256Digest;
import org.bouncycastle.crypto.encodings.OAEPEncoding;
import org.bouncycastle.crypto.engines.RSAEngine;
import org.bouncycastle.crypto.params.RSAKeyParameters;

import java.io.ByteArrayOutputStream;
import java.io.IOException;

public class Encrypt {
    public static byte[] EncryptBytes(RSAKeyParameters publicKeyParam, byte[] content) throws InvalidCipherTextException, IOException {
        RSAEngine rsaEngine = new RSAEngine();
        OAEPEncoding oaepEncoding = new OAEPEncoding(rsaEngine, new SHA256Digest(), new SHA256Digest(), null);
        oaepEncoding.init(true, publicKeyParam);
        int inputBlockSize = oaepEncoding.getInputBlockSize();
        ByteArrayOutputStream bo = new ByteArrayOutputStream();

        for(int i=0;i<content.length;i+=inputBlockSize){
            byte[] cipherText = oaepEncoding.processBlock(content, i, i + inputBlockSize < content.length ? inputBlockSize: content.length - i);
            bo.write(cipherText);
        }
        bo.flush();
        return bo.toByteArray();
    }

    public static String EncryptBytesAndEncodeWithBase64(RSAKeyParameters publicKeyParam, byte[] content) throws InvalidCipherTextException, IOException {
        byte[] cipherBytes = EncryptBytes(publicKeyParam, content);
        return org.bouncycastle.util.encoders.Base64.toBase64String(cipherBytes);
    }
}
